tests/integration_tests.rs

```diff
@@ -39,6 +41,73 @@ fn bat() -> assert_cmd::Command {
     cmd
 }
 
+/// For some tests we want mocked versions of some pagers
+/// This fn returns the absolute path to the directory with these mocked pagers
+fn get_mocked_pagers_dir() -> PathBuf {
+    let cargo_manifest_dir = env::var("CARGO_MANIFEST_DIR").expect("Missing CARGO_MANIFEST_DIR");
+    Path::new(&cargo_manifest_dir)
+        .join("tests")
+        .join("mocked-pagers")
+}
+
+/// On Unix: 'most' -> 'most'
+/// On Windows: 'most' -> 'most.bat'
+fn mocked_pager(base: &str) -> String {
+    if cfg!(windows) {
+        let mut str = String::from(base);
+        str.push_str(".bat");
+        str
+    } else {
+        String::from(base)
+    }
+}
+
+/// Prepends a directory to the PATH environment variable
+/// Returns the original value for later restoration
+fn prepend_dir_to_path_env_var(dir: PathBuf) -> String {
+    // Get current PATH
+    let original_path = env::var("PATH").expect("No PATH?!");
+
+    // Add the new dir first
+    let mut split_paths = env::split_paths(&original_path).collect::<Vec<_>>();
+    split_paths.insert(0, dir);
+
+    // Set PATH with the new dir
+    let new_path = env::join_paths(split_paths).expect("Failed to join paths");
+    env::set_var("PATH", new_path);
+
+    // Return the original value for later restoration of it
+    original_path
+}
+
+/// Helper to restore the value of PATH
+fn restore_path(original_path: String) {
+    env::set_var("PATH", original_path);
+}
+
+/// Allows test to run that require our mocked versions of 'more' and 'most'
+/// in PATH. Temporarily changes PATH while the test code runs, and then restore it
+/// to avoid pollution of global state
+fn with_mocked_versions_of_more_and_most_in_path(actual_test: fn()) {
```

Comment:
I love this :+1:. It would be nice to move all of these functions into a separate file, I think. An alternative idea for implementation might be to use a `struct WithMockedVersionsOfMoreAndMostInPath` that has a `run(actual_test fn())` member function and a `impl Drop` that resets the PATH. It could store the original path as a member variable. This way, one could run multiple tests in sequence.

But I'm not sure if that really improves the code. The current implementation is also completely fine.

Reply:
Using a struct with Drop is an interesting idea, I'll try it out and change to that if it ends up feeling and looking and being better

I'll also move that stuff out into a separate file 👍